app/routes/_welcome+/onboarding.tsx

```diff
@@ -133,6 +136,17 @@ export async function action({ context, request }: ActionFunctionArgs) {
      * This is to make sure that we have a stripe customer for the user.
      * We have to do it at this point, as its the first time we have the user's first and last name
      */
+
+    if (user) {
```

Comment:
I don't understand this? Why wouldn't there be a session already? That should not happen, this is a protected route. If it happens that there is no session it needs to be resolved properly by finding whats the cause of the issue.

Reply:
> I don't understand this? Why wouldn't there be a session already? That should not happen, this is a protected route. If it happens that there is no session it needs to be resolved properly by finding whats the cause of the issue.

In Supabase, updating a user's password using the updateUserById function will invalidate the user's current session, logging them out as part of security best practices. If you need to update a user's password without invalidating their session, you will have to handle it with caution due to potential security risks. However, here's a general approach you might consider:

Reply:
> > I don't understand this? Why wouldn't there be a session already? That should not happen, this is a protected route. If it happens that there is no session it needs to be resolved properly by finding whats the cause of the issue.
> 
> In Supabase, updating a user's password using the updateUserById function will invalidate the user's current session, logging them out as part of security best practices. If you need to update a user's password without invalidating their session, you will have to handle it with caution due to potential security risks. However, here's a general approach you might consider:

Can you share the doc about that? Maybe I miss something (or this is something new)

Reply:
I also can't find it in the docs, but actually when I think about it, when they reset their password, we shouldn't automatically log them in. We should actually just send them to the login page so they can login with their new pwd. I think that is better for security and it will completely resolve the issue we are having here.

Reply:
> I also can't find it in the docs, but actually when I think about it, when they reset their password, we shouldn't automatically log them in. We should actually just send them to the login page so they can login with their new pwd. I think that is better for security and it will completely resolve the issue we are having here.

I have implemented it without breaking the existing flow. Again it's your call how the flow should be.